### PR TITLE
[bitnami/flux] fix: :lock: Move service-account token auto-mount to pod declaration

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier Salmeron Garcia <jsalmeron@vmware.com>

### Description of the change

This PR sets all ServiceAccount automountServiceAccountToken=false by default, as the token mounting should be set in the pod declaration instead (if a new pod uses this service account and the token gets automatically mounted, it could be problematic in terms of security). This PR also adds automountServiceAccountToken in the pod declaration as a new value, which can be configured by users in case they want to use an external token.

### Benefits

Charts become more security-compliant

### Possible drawbacks

n/a

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/flux/values.yaml [hunk 1]
```diff
@@ -218,6 +218,9 @@ kustomizeController:
   ## @param kustomizeController.args Override default container args (useful when using custom images)
   ##
   args: []
+  ## @param kustomizeController.automountServiceAccountToken Mount Service Account token in pod
+  ##
+  automountServiceAccountToken: true
```
Comment (joancafom) [inline bitnami/flux/values.yaml:223]: shouldn't it be set to `false` instead?

File: bitnami/flux/values.yaml [hunk 2]
```diff
@@ -664,6 +667,9 @@ helmController:
   ## @param helmController.args Override default container args (useful when using custom images)
   ##
   args: []
+  ## @param helmController.automountServiceAccountToken Mount Service Account token in pod
+  ##
+  automountServiceAccountToken: true
```
Comment (joancafom) [inline bitnami/flux/values.yaml:672]: shouldn't it be set to `false` instead?

File: bitnami/flux/values.yaml [hunk 3]
```diff
@@ -1112,6 +1118,9 @@ sourceController:
   ## @param sourceController.args Override default container args (useful when using custom images)
   ##
   args: []
+  ## @param sourceController.automountServiceAccountToken Mount Service Account token in pod
+  ##
+  automountServiceAccountToken: true
```
Comment (joancafom) [inline bitnami/flux/values.yaml:1123]: shouldn't it be set to `false` instead?

File: bitnami/flux/values.yaml [hunk 4]
```diff
@@ -1716,6 +1725,9 @@ notificationController:
   ## @param notificationController.args Override default container args (useful when using custom images)
   ##
   args: []
+  ## @param notificationController.automountServiceAccountToken Mount Service Account token in pod
+  ##
+  automountServiceAccountToken: true
```
Comment (joancafom) [inline bitnami/flux/values.yaml:1730]: shouldn't it be set to `false` instead?

File: bitnami/flux/values.yaml [hunk 5]
```diff
@@ -2162,6 +2174,9 @@ imageAutomationController:
   ## @param imageAutomationController.args Override default container args (useful when using custom images)
   ##
   args: []
+  ## @param imageAutomationController.automountServiceAccountToken Mount Service Account token in pod
+  ##
+  automountServiceAccountToken: true
```
Comment (joancafom) [inline bitnami/flux/values.yaml:2179]: shouldn't it be set to `false` instead?

File: bitnami/flux/values.yaml [hunk 6]
```diff
@@ -2608,6 +2623,9 @@ imageReflectorController:
   ## @param imageReflectorController.args Override default container args (useful when using custom images)
   ##
   args: []
+  ## @param imageReflectorController.automountServiceAccountToken Mount Service Account token in pod
+  ##
+  automountServiceAccountToken: true
```
Comment (joancafom) [inline bitnami/flux/values.yaml:2628]: shouldn't it be set to `false` instead?

File: bitnami/flux/values.yaml (hunk 1 above)
Comment (javsalgar) [reply bitnami/flux/values.yaml:223]: They require access to the K8s API so they need to mount the SA token

File: bitnami/flux/values.yaml (hunk 2 above)
Comment (javsalgar) [reply bitnami/flux/values.yaml:672]: They require access to the K8s API so they need to mount the SA token

File: bitnami/flux/values.yaml (hunk 3 above)
Comment (javsalgar) [reply bitnami/flux/values.yaml:1123]: They require access to the K8s API so they need to mount the SA token

File: bitnami/flux/values.yaml (hunk 4 above)
Comment (javsalgar) [reply bitnami/flux/values.yaml:1730]: They require access to the K8s API so they need to mount the SA token

File: bitnami/flux/values.yaml (hunk 5 above)
Comment (javsalgar) [reply bitnami/flux/values.yaml:2179]: They require access to the K8s API so they need to mount the SA token

File: bitnami/flux/values.yaml (hunk 6 above)
Comment (javsalgar) [reply bitnami/flux/values.yaml:2628]: They require access to the K8s API so they need to mount the SA token

**Review by joancafom — APPROVED** [review]

**Pull request merged**